openslides/users/views.py

```diff
@@ -165,6 +169,74 @@ def mass_import(self, request):
             'detail': _('{number} users successfully imported.').format(number=len(created_users)),
             'importedTrackIds': imported_track_ids})
 
+    @list_route(methods=['post'])
+    def mass_invite_email(self, request):
+        user_ids = request.data.get('user_ids')
+        if not isinstance(user_ids, list):
+            raise ValidationError({'detail': 'User_ids has to be a list.'})
+        users = []
+        for user_id in user_ids:
+            if not isinstance(user_id, int):
+                raise ValidationError({'detail': 'User_id has to be an int.'})
+            try:
+                users.append(User.objects.get(pk=user_id))
+            except User.DoesNotExist:
+                raise ValidationError({'detail': 'The user with id {} does not exist'.format(user_id)})
+
+        # Sending Emails. Keep track, which users gets an email.
+        success_users = []
+        connection = mail.get_connection(fail_silently=False)
+        try:
+            connection.open()
+        except ConnectionRefusedError:
+            return Response({'detail': 'Cannot connect to SMTP server on {}:{}'.format(
+                settings.EMAIL_HOST,
+                settings.EMAIL_PORT)}, status=400)
+        except smtplib.SMTPException as e:
+            return Response({'detail': '{}: {}'.format(e.errno, e.strerror)}, status=400)
+
+        for user in users:
```

Comment:
Why not use https://docs.djangoproject.com/en/1.10/topics/email/#send-mass-mail? It is slightly more efficient.

Reply:
With this approach we do have more control: We can check, if the email for each users was send, because send_mass_email just returns the email count. if 8 out of 10 emails were send, we cannot keep track, which user does not get an email.

Also this should be equally efficient in fact that both cases just opens one connection and try to send the same amount of emails - in our case with the ability to see which email was successfully send.

Reply:
Ok.